app/src/test/java/org/mozilla/fenix/components/BackgroundServicesTest.kt

```diff
@@ -35,101 +39,101 @@ class BackgroundServicesTest {
-
-        // Other external
+        verify { metrics.track(Event.SyncAuthRecovered) }
+        verify { settings.fxaSignedIn = true }
+        confirmVerified(metrics, settings)
+    }
+
+    @Test
+    fun `telemetry account observer tracks external creation event with null action`() {
+        val account = mockk<OAuthAccount>()
+
         registry.notifyObservers { onAuthenticated(account, AuthType.OtherExternal(null)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthSignIn)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthSignUp)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthPaired)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthFromShared)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthRecovered)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthOtherExternal)) }
-        verify(exactly = 0) { metrics.track(eq(Event.SyncAuthSignOut)) }
+        verify { metrics.track(Event.SyncAuthOtherExternal) }
+        verify { settings.fxaSignedIn = true }
+        confirmVerified(metrics, settings)
+    }
+
+    @Test
+    fun `telemetry account observer tracks external creation event with some action`() {
+        val account = mockk<OAuthAccount>()
 
         registry.notifyObservers { onAuthenticated(account, AuthType.OtherExternal("someAction")) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthSignIn)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthSignUp)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthPaired)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthFromShared)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthRecovered)) }
-        verify(exactly = 2) { metrics.track(eq(Event.SyncAuthOtherExternal)) }
-        verify(exactly = 0) { metrics.track(eq(Event.SyncAuthSignOut)) }
-
-        // NB: 'Existing' auth type isn't expected to record any auth telemetry.
+        verify { metrics.track(Event.SyncAuthOtherExternal) }
+        verify { settings.fxaSignedIn = true }
+        confirmVerified(metrics, settings)
+    }
+
+    @Test
+    fun `telemetry account observer does not track existing account`() {
+        val account = mockk<OAuthAccount>()
+
         registry.notifyObservers { onAuthenticated(account, AuthType.Existing) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthSignIn)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthSignUp)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthPaired)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthFromShared)) }
-        verify(exactly = 1) { metrics.track(eq(Event.SyncAuthRecovered)) }
-        verify(exactly = 2) { metrics.track(eq(Event.SyncAuthOtherExternal)) }
-        verify(exactly = 0) { metrics.track(eq(Event.SyncAuthSignOut)) }
-
-        // Logout
+        verify(inverse = true) { metrics.track(Event.SyncAuthSignIn) }
```

Comment:
Would `verify { metrics wasNot Called }` work here?

Reply:
I tried a variation of that and I couldn't figure out what was causing the issue. Let me double-check.